docs/proposals/13-automated-seed-management.md

```diff
@@ -0,0 +1,338 @@
+    metadata:
+      labels:
+        foo: bar
+    spec:
+      provider:
+        type: gcp
+        region: europe-west1
+      taints:
+      - key: seed.gardener.cloud/protected
+      ...
+```
+
+Example `ManagedSeed` resource with a `gardenlet` section:
+
+```yaml
+apiVersion: seedmanagement.gardener.cloud/v1alpha1
+kind: ManagedSeed
+metadata:
+  name: crazy-botany
+  namespace: garden
+spec:
+  shootName: crazy-botany # Shoot that should be registered as a Seed
+  gardenlet: 
+    deployment: # Gardenlet deployment configuration
+      replicas: 1
+      revisionHistoryLimit: 10
+      serviceAccountName: gardenlet
+      image:
+        repository: eu.gcr.io/gardener-project/gardener/gardenlet
+        tag: latest
+        pullPolicy: IfNotPresent
+      resources:
+        ...
+      annotations: 
+        ...
+      labels:
+        ...
+      env:
+        ...
+      imageVectorOverwrite: | # See docs/deployment/image_vector.md
+        ...
+      componentImageVectorOverwrites: | # See docs/deployment/image_vector.md
+        ...
+      dnsConfig:
+        ...
+      additionalVolumes:
+        ...
+      additionalVolumeMounts:
+        ...
+      vpa: false
+    config: # GardenletConfiguration resource
+      seedConfig: # Configuration of the Seed to be registered
+        ...
+      controllers:
+        shoot:
+          concurrentSyncs: 20
+      featureGates:
+        CachedRuntimeClients: true
+      ...
+```
+
+### ManagedSeed Controller
+
+ManagedSeeds are reconciled by a new *managed seed controller* in `gardenlet`. Its implementation is very similar to the current [seed registration controller](https://github.com/gardener/gardener/blob/master/pkg/gardenlet/controller/shoot/seed_registration_control.go), and in fact could be regarded as a refactoring of the latter, with the difference that it uses the `ManagedSeed` resource rather than the `use-as-seed` annotation on a Shoot. The `gardenlet` only reconciles ManagedSeeds that refer to Shoots scheduled on Seeds the `gardenlet` is responsible for.
+
+Once this controller is considered sufficiently stable, the current `use-as-seed` annotation and the controller mentioned above should be marked as deprecated and eventually removed.
+
+A `ManagedSeed` that is in use by shoots cannot be deleted, unless the shoots are either deleted or moved to other seeds first. The managed seed controller ensures that this is the case.
+
+**Note:** Bootstrapping a new seed might require additional provider-specific actions to the ones performed automatically by the managed seed controller. For example, on Azure this might include getting a new subscription, extending quotas, etc. This could eventually be automated by introducing an extension mechanism for the Gardener seed bootstrapping flow, to be handled by a new type of controller in the provider extensions. However, such an extension mechanism is not in the scope of this proposal and might require a separate GEP.
```

Comment:
Can you just briefly mention the idea with the pod readiness gates here so that we don't forget it?

Reply:
Converted the note into a section and added two more sentences to briefly outline the idea.